pkg/conditions/manager.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2025 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package conditions enables consistent interactions with an object's status conditions.
+package conditions
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+// ObjectWithConditions is the interface definition that allows.
+type ObjectWithConditions interface {
+	resource.Object
+	resource.Conditioned
+}
+
+// Manager is an interface for a stateless factory-like object that produces ConditionSet objects.
+type Manager interface {
+	// For returns an implementation of a ConditionSet to operate on a specific ObjectWithConditions.
+	For(o ObjectWithConditions) ConditionSet
+}
+
+// ConditionSet holds operations for interacting with an object's conditions.
+type ConditionSet interface {
+	// Mark adds or updates the conditions onto the managed resource object. Unlike a "Set" method, this also can add
+	// contextual updates to the condition such as propagating the correct observedGeneration to the conditions being
+	// changed.
+	Mark(condition ...xpv1.Condition)
```

Comment:
WDYT about `MarkConditions`?

While I generally like the UX for this, looking at e.g. https://github.com/crossplane/crossplane-runtime/pull/828/files#diff-97300a2543f95f5a2ada3560bf47dd7334e237e27976574d15d1cddef2e66c01L883 `status.MarkConditions(Available())` would give me more of a hint to what it's doing than `status.Mark(Available()`.

I think I'd prefer to just stick with `SetConditions` despite the extra magic happening behind the scenes, but I don't feel as strongly about that.

Reply:
I will change this to `MarkConditions`, I think something that is not `Set` makes it clear there is magic happening or at least clear it is not a normal `Getter/Setter` pattern.